app/main.rs

```diff
@@ -66,12 +66,52 @@ fn main() {
             let this_branch_directory: Directory = store
                 .read_json(this_snapshot.directory)
                 .expect("read this branch directory");
-            // TODO Format this more nicely, this is cringe
-            serde_json::to_writer_pretty(
-                stdout(),
-                &this_branch_directory.diff(&that_branch_directory),
-            )
-            .unwrap();
+            let diff = &this_branch_directory.diff(&that_branch_directory);
+
+            fn go_added(path: &str, entry: &DirectoryEntry) {
```

Comment:
Should probably use `PathBuf` https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html

Reply:
Also this could probably be a method on `Diff`, called like `diff.print()` or something? I don't know what's idiomatic.

Reply:
I think maybe `impl Display for Diff { ... }` would be nice here, cause then it could be exposed as an idiomatic library as well.